qutebrowser/browser/commands.py

```diff
@@ -822,22 +876,36 @@ def zoom(self, zoom: int = None, count=None):
         message.info("Zoom level: {}%".format(level), replace=True)
 
     @cmdutils.register(instance='command-dispatcher', scope='window')
-    def tab_only(self, prev=False, next_=False):
+    def tab_only(self, prev=False, next_=False, force=False):
         """Close all tabs except for the current one.
 
         Args:
             prev: Keep tabs before the current.
             next_: Keep tabs after the current.
+            force: Avoid confirmation for pinned tabs.
         """
         cmdutils.check_exclusive((prev, next_), 'pn')
         cur_idx = self._tabbed_browser.currentIndex()
         assert cur_idx != -1
 
+        def _to_close(i):
+            """Helper method to check if a tab should be closed or not."""
+            return not (i == cur_idx or
+                        (prev and i < cur_idx) or
+                        (next_ and i > cur_idx))
+
+        # Check to see if we are closing any pinned tabs
```

Comment:
The reason for this is to prevent calling `close_tab` if we have any pinned tabs at all, That way, if someone runs `:tab-only` either the action is completed (all other tabs are closed), or no tabs are closed at all (at least one pinned tab found). I don't think it's good (user-wise) to have a situation where we close as many tabs as we can on the right, and stop when we see a tab we cannot close.